### DISCUSSION TITLE
Unable to connect to Milvus using attu on windows

### DISCUSSION THREAD

**Discussion by luliloveshe** [milvus-io/milvus#27942] [Q&A and General discussion]
There was no response when clicking the attu "connect" button. I uploaded the log file and took screenshots of some error areas in the log. In addition, there is ip169.254.248.102 in the log, which is not the IP address I used.

I successfully started the same binary package on my local Windows 10 machine. If the startup is successful, the log will contain the string "---Milvus Proxy successfully initialized and ready to serve!---". If the startup fails, there will be no string.

![{7%{NIWQAK3 MI)J7JL5($A](https://github.com/milvus-io/milvus/assets/32316248/47335138-5690-43dc-a6e3-4c4799846bb2)
![UV5~BOZGC10)%NC3H}037SV](https://github.com/milvus-io/milvus/assets/32316248/29dd3893-cd27-4475-8ee6-649356e6a0c4)
![~YHWRTMQVC32%MK GO~BU H](https://github.com/milvus-io/milvus/assets/32316248/6a7382cf-0bdf-47fa-9bc2-e35f83703013)
![M`U}EYERC%F`LKPK) ZK9OL](https://github.com/milvus-io/milvus/assets/32316248/a081f54f-a940-4e97-94eb-4279303b7c77)
[log.txt](https://github.com/milvus-io/milvus/files/13174121/log.txt)

**Comment by yhmo** [comment]
All the components are successfully started, including the rootCoord:
`[2023/10/26 15:08:59.187 +08:00] [DEBUG] [components/root_coord.go:61] ["RootCoord successfully started"]`

But other components still cannot connect the rootCoord:
`[2023/10/26 15:08:59.349 +08:00] [WARN] [grpclog/grpclog.go:46] ["[core][Channel #152 SubChannel #153] grpc: addrConn.createTransport failed to connect to {\n  \"Addr\": \"169.254.248.102:53100\"`

I believe this is not an official build.
```
Welcome to use Milvus!
Version:   windows-v2.2.0
Built:     Thu, Dec  8, 2022 10:29:07 AM
GitCommit: d54acb8
GoVersion: go version go1.18 windows/amd64
```
I didn't see the commit **d54acb8** in the commit history, I guess you have changed the code. I have no idea why it failed.

**Reply by luliloveshe** [reply]
Due to historical reasons, our machines are all windows and we can only run the standalone version of Milvus on windows. Refer to https://milvus.io/blog/2021-11-19-run-milvus-2.0-on-windows.md#Install-from-the-binary-package. I didn't modify the code myself.
![0WY08N6YU3GRGJ_X NQHVNA](https://github.com/milvus-io/milvus/assets/32316248/73082a05-10f2-45c1-ae7e-8777daa60525)
![HG3%U812{IJ Q)88JBK9}SY](https://github.com/milvus-io/milvus/assets/32316248/341d12a3-07af-4ce8-bff8-e93c829f2d25)
![U09OL)0)`W8W5K%ZTKE%_1H](https://github.com/milvus-io/milvus/assets/32316248/909d2bb6-6f8f-4ea2-adb8-7aeb7f5a086f)

**Reply by yhmo** [reply]
Hmmm... we don't have much experience on Windows. You can contact @matrixji for more information.

**Comment by matrixji** [comment]
While running the v2.2.0 Windows binary, it may use the 169.254.x.x loopback interface for internal communications, but actually, it could not be connected with that address.
I do a dedicated patch for it on milvus-lite, ref: https://github.com/milvus-io/milvus-lite/blob/8bd10fc577cac122fb5d94ce25fa6cab8ec8bbf4/milvus_binary/patches/milvus-v2.2.14/0001-patch-for-milvus-lite.patch#L617C12-L617C12
Milvus-lite could run under Windows until v2.2.14. (v2.3 for Windows is not supported right now)
I prefer to use [milvus-lite](https://github.com/milvus-io/milvus-lite/) under Windows.  And you could also extract the Windows binary from it.

ref example commands:
```
pip install milvus
milvus-server -d d:\milvus_dir
```
The milvus server is started. After you stop the milvus server(Ctrl+C), you can find the milvus binary under d:\milvus_dir (may change any location you want)

**Reply by luliloveshe** [reply]
Thank you very much for the answer. In addition, I would like to ask if binary deployment on Windows can be used in a production environment. Because I deployed the binary to other machines Windows Server 2019, it started successfully. But as soon as two vectors were inserted, an error was reported. After restarting, it was not queried on attu.

**Reply by matrixji** [reply]
Not suggested to use milvus windows for Production use. It only supports standalone mode with single node, and still not full tested
